src/stores/EndpointConfig/hooks.ts

```diff
@@ -1,229 +1,98 @@
-import { useEntityType } from 'context/EntityContext';
-import { useZustandStore } from 'context/Zustand/provider';
-import { EndpointConfigStoreNames } from 'stores/names';
-import { Entity } from 'types';
-import { EndpointConfigState } from './types';
-
-const getStoreName = (entityType: Entity): EndpointConfigStoreNames => {
-    if (entityType === 'capture' || entityType === 'materialization') {
-        return EndpointConfigStoreNames.GENERAL;
-    } else {
-        throw new Error('Invalid EndpointConfig store name');
-    }
-};
+import { useShallow } from 'zustand/react/shallow';
+import { useEnpointConfigStore } from './Store';
 
 export const useEndpointConfigStore_errorsExist = () => {
-    const entityType = useEntityType();
-
-    return useZustandStore<
-        EndpointConfigState,
-        EndpointConfigState['errorsExist']
-    >(getStoreName(entityType), (state) => state.errorsExist);
+    return useEnpointConfigStore((state) => state.errorsExist);
 };
 
 export const useEndpointConfigStore_endpointConfigErrors = () => {
-    const entityType = useEntityType();
-
-    return useZustandStore<
-        EndpointConfigState,
-        EndpointConfigState['endpointConfigErrors']
-    >(getStoreName(entityType), (state) => state.endpointConfigErrors);
+    return useEnpointConfigStore((state) => state.endpointConfigErrors);
 };
 
 export const useEndpointConfigStore_reset = () => {
-    const entityType = useEntityType();
-
-    return useZustandStore<
-        EndpointConfigState,
-        EndpointConfigState['resetState']
-    >(getStoreName(entityType), (state) => state.resetState);
+    return useEnpointConfigStore((state) => state.resetState);
 };
 
 export const useEndpointConfigStore_changed = () => {
-    const entityType = useEntityType();
-
-    return useZustandStore<
-        EndpointConfigState,
-        EndpointConfigState['stateChanged']
-    >(getStoreName(entityType), (state) => state.stateChanged);
+    return useEnpointConfigStore(useShallow((state) => state.stateChanged));
```

Comment:
Threw this tweak in while cleaning up. Since this is a computed value I think we're better off using shallow here.